### ISSUE TITLE
Labels in ATX-style headings which contain a prefix ending in a colon have this stripped

### ISSUE THREAD

**Issue by robwalton** [shd101wyy/mume#46]
I've tracked down an issue that effects the atom and vscode MPE pandoc preview system to the mume library. When these applications use mume during preview to convert a markdown file before sending its contents to pandoc there is a corner case in which mume breaks some ATX-style header labels.

For example when previewing the one line markdown file in MPE:
```markdown
# testing {#sec:testing}
```
mume removes the `sec:` part of the label:
```markdown
<p data-line="0" class="sync-line" style="margin:0;"></p>

# testing {#testing }
```
This is not a problem when using setext-style headers (that is with underlines). This causes the popular [pandoc-crossref](http://lierdakil.github.io/pandoc-crossref/) filter, which requires section labels to begin with 'sec:', to break.

Sorry for the ticket spam! I created shd101wyy/markdown-preview-enhanced#816 with the symptoms and then shd101wyy/vscode-markdown-preview-enhanced#83 once I got closer to the cause.

**Comment by robwalton** [comment]
I've fired it up in the vscode debugger and discovered that removing the ':' from [this line in attributes/parse.ts](https://github.com/shd101wyy/mume/blob/949135eb77de24f373fa5fd4ea1becf2ed060899/src/lib/attributes/parse.ts#L101):  fixes the problem. When I get a chance to get the tests running (not a javascript guy!) I'll make a pull request.

**Comment by robwalton** [comment]
Since you are here @shd101wyy. I've got mume checked out and built in vscode (I see 'Starting compilation in watch mode...'). Any hints on how to run the tests? (I'm sure obvious to typescript/javascript people!)

**Comment by kachkaev** [comment]
Thanks @robwalton! It'd be great if you could add a failing test first, then fix the implementation and finally make sure that the new test passes. Test cases for `attributes` are in [test/lib/attributes.test.ts](https://github.com/shd101wyy/mume/blob/master/test/lib/attributes.test.ts).

**Comment by robwalton** [comment]
Thanks @kachkaev I found those. Any hints on how to run them. Sorry, I don't know about typescript or javascript things.

**Comment by kachkaev** [comment]
You can start tests by running `npm run test`. You can also try `npm run test --watch` if you want to run the tests continiously. See jest docs for more options https://facebook.github.io/jest/docs/en/cli.html

I'm happy to help if you have more questions.

**Comment by robwalton** [comment]
Awesome, thanks. I got the tests running. I'll give it a go later today.

**Comment by robwalton** [comment]
I've added some failing tests to reflect the issue here and the fix suggested earlier fixes these. However it breaks three tests. I'm quite happy to find a way not to break these. However, before I going adding complexity, I wanted to make sure that the failing tests are important.

They are the first two in the suite and the only to use `cmd:true` to specify a key-value pair rather than`cmd=true`. I'm pretty sure this is not required by pandoc and its definitely not part of the standard for headers described in the [pandoc manual](https://pandoc.org/MANUAL.html#header-identifiers), but maybe there is another use I'm not aware of for this functionality?

Two [here](https://github.com/shd101wyy/mume/blob/949135eb77de24f373fa5fd4ea1becf2ed060899/test/lib/attributes.test.ts#L20):
```typescript
  {
    // classic behavior
    attributes: { cmd: true },
    raw: [
      "cmd=true",
      "{cmd=true}",
      "  {  cmd=true  }  ",
      "{cmd:true}",       // FAILING
      "cmd:true",        // FAILING
    ],
    stringified: "cmd=true",
  },

```
And one [here](https://github.com/shd101wyy/mume/blob/949135eb77de24f373fa5fd4ea1becf2ed060899/test/lib/attributes.test.ts#L27)
```typescript
  {
    attributes: { cmd: true, hello: "world" },
    raw: ["cmd=true hello=world", "cmd:true hello:world"],  // SECOND FAILING
    stringified: 'cmd=true hello="world"',
  },
```
Do we need to support these tests? (If in doubt I will.)

**Comment by kachkaev** [comment]
I added `:` as a synonym of `=` in https://github.com/shd101wyy/mume/commit/4898f1f1b077a725b0fa26f3830a06db51448357 after noticing its use in [TOC](https://shd101wyy.github.io/markdown-preview-enhanced/#/toc?id=configuration) (see first screenshot).

I'm happy to remove `:` as a synonym to `=` but I'll let @shd101wyy decide.

**Comment by shd101wyy** [comment]
@kachkaev @robwalton  Sure we could remove it. Thank you ;)

**Comment by kachkaev** [comment]
@robwalton feel free to roll back https://github.com/shd101wyy/mume/commit/4898f1f1b077a725b0fa26f3830a06db51448357 and add a test  for you case to make sure we don't break it in future. It'd be great to remove any mentioning of `:` as `=` from the docs too!

**Comment by robwalton** [comment]
I've submitted a pull request. @kachkaev 

> It'd be great to remove any mentioning of : as = from the docs too!

I had a quick look and not sure where you mean on these?

**Comment by kachkaev** [comment]
The reason why I added `:` as a synonym for `=` in https://github.com/shd101wyy/mume/commit/4898f1f1b077a725b0fa26f3830a06db51448357 was because I saw `:` in the docs (screenshot 1 on [TOC page](https://shd101wyy.github.io/markdown-preview-enhanced/#/toc?id=configuration) in particular). Give that colons are no longer supported as a delimiter between the key and the value, it'd be great if this was reflected in the dos. @shd101wyy can you recall in what parts of it you could refer to `:` as `=`?

**Comment by shd101wyy** [comment]
@kachkaev Yes we will need to update the doc. The toc is defined as a code chunk. You probably could refer to [this line](https://github.com/shd101wyy/mume/blob/949135eb77de24f373fa5fd4ea1becf2ed060899/src/render-enhancers/fenced-code-chunks.ts#L209). Thank you.

**Comment by shd101wyy** [comment]
I just updated the documentation for TOC [here](https://shd101wyy.github.io/markdown-preview-enhanced/#/toc). Feel free to remove `:` synonym. Thank you.

**Comment by kachkaev** [comment]
@shd101wyy that should be done by @robwalton in #48. The PR LGTM – feel free to review it too and merge.

**Comment by robwalton** [comment]
Hi guys, sorry for the delay. It would have been cleaner for me to include the doc change in my pull request. Still happy to do so, but looks like its all moved forward now.

**Comment by kachkaev** [comment]
@shd101wyy it'd be great if you could look at https://github.com/shd101wyy/mume/pull/48 as well as two other PRs from me and cut a new release 🎉

**Comment by shd101wyy** [comment]
@kachkaev Sure I will do it this weekend and publish a new version. Thank you ;)
